extensions/src/AWSSDK.Extensions.Configuration.AWSSystemsManager/AWSSystemsManagerExtensions.cs

```diff
@@ -0,0 +1,85 @@
+        /// </summary>
+        /// <param name="builder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="awsOptions"><see cref="AWSOptions"/> used to create an AWS Systems Manager Client connection</param>
+        /// <param name="path">The path that variable names must start with. The path will be removed from the variable names.</param>
+        /// <param name="optional">Whether the AWS Systems Manager Parameters are optional.</param>
+        /// <param name="reloadAfter">Initiate reload after TimeSpan</param>
+        /// <param name="onLoadException">Delegate to call on Exception</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddAWSSystemsManager(this IConfigurationBuilder builder, AWSOptions awsOptions, string path, bool optional = false, TimeSpan? reloadAfter = null, Action<AWSSystemsManagerExceptionContext> onLoadException = null)
+        {
+            if (string.IsNullOrWhiteSpace(path)) throw new ArgumentNullException(nameof(path));
+            if (awsOptions == null) throw new ArgumentNullException(nameof(awsOptions));
+            return builder.AddAWSSystemsManager(source =>
+            {
+                source.AwsOptions = awsOptions;
+                source.Path = path;
+                source.Optional = optional;
+                source.ReloadAfter = reloadAfter;
+                source.OnLoadException = onLoadException;
+            });
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from AWS Systems Manager Parameter variables with a specified path.
+        /// </summary>
+        /// <param name="builder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="path">The path that variable names must start with. The path will be removed from the variable names.</param>
+        /// <param name="optional">Whether the AWS Systems Manager Parameters are optional.</param>
+        /// <param name="reloadAfter">Initiate reload after TimeSpan</param>
+        /// <param name="onLoadException">Delegate to call on Exception</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddAWSSystemsManager(this IConfigurationBuilder builder, string path, bool optional = false, TimeSpan? reloadAfter = null, Action<AWSSystemsManagerExceptionContext> onLoadException = null)
+        {
+            if (string.IsNullOrWhiteSpace(path)) throw new ArgumentNullException(nameof(path));
+            var config = builder.Build();
+            return builder.AddAWSSystemsManager(config.GetAWSOptions(), path, optional, reloadAfter, onLoadException);
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from environment variables.
```

Comment:
Incorrect code documentation.

Reply:
Good catch. I copied the hints from another implementation, but I guess I missed updating this text